pkg/apis/provisioning/v1alpha5/taints.go

```diff
@@ -44,17 +44,31 @@ func (ts Taints) HasKey(taintKey string) bool {
 	return false
 }
 
-// Tolerates returns true if the pod tolerates all taints
-func (ts Taints) Tolerates(pod *v1.Pod) (errs error) {
+// Tolerates returns true if the pod tolerates all taints. 'additional' are extra tolerations for things like startup
+// taints or the standard not-ready taint applied to nodes we have launched.
+func (ts Taints) Tolerates(pod *v1.Pod, additional ...v1.Toleration) (errs error) {
 	for i := range ts {
 		taint := ts[i]
 		tolerates := false
 		for _, t := range pod.Spec.Tolerations {
 			tolerates = tolerates || t.ToleratesTaint(&taint)
 		}
+		for _, t := range additional {
+			tolerates = tolerates || t.ToleratesTaint(&taint)
+		}
 		if !tolerates {
 			errs = multierr.Append(errs, fmt.Errorf("did not tolerate %s=%s:%s", taint.Key, taint.Value, taint.Effect))
 		}
 	}
 	return errs
 }
+
+// TaintToToleration converts a taint to a toleration that tolerates the specified taint.
+func TaintToToleration(taint v1.Taint) v1.Toleration {
```

Comment:
Many tolerations can tolerate the same taint. Should we make the toleration just tolerate the key, not necessarily the value? We also just want NoScheduler here in all cases, right?

Reply:
I wrote it this way so that if the node taint isn't exactly what we created, it doesn't get the special behavior.  I think that's the safest way to start this.  We place the taint on the node originally, so it will match.  If the taint then changes in some way, I don't think we can safely assume that we should still treat it as a startup taint.